tpr/tpr.go

```diff
@@ -0,0 +1,120 @@
+package tpr
+
+import (
+	"fmt"
+	"time"
+
+	microerror "github.com/giantswarm/microkit/error"
+	"github.com/rossf7/pingdom-operator/pkg/util"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/errors"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	tprInitRetries    = 30
+	tprInitRetryDelay = 3 * time.Second
+)
+
+type Config struct {
+	// Dependencies.
+	Clientset kubernetes.Interface
+	Rest      rest.Interface
+
+	// Settings.
+	Name        string
+	Group       string
+	Version     string
+	Description string
+}
+
+type tpr struct {
+	clientset kubernetes.Interface
+
+	name          string
+	group         string
+	version       string
+	description   string
+	qualifiedName string
+
+	endpointList string
+}
+
+func New(config Config) (*tpr, error) {
+	if config.Clientset == nil {
+		return nil, microerror.MaskAnyf(invalidConfigError, "k8s clientset must be set")
+	}
+	if config.Name == "" {
+		return nil, microerror.MaskAnyf(invalidConfigError, "name must not be empty")
+	}
+	if config.Group == "" {
+		return nil, microerror.MaskAnyf(invalidConfigError, "group must not be empty")
+	}
+	if config.Version == "" {
+		return nil, microerror.MaskAnyf(invalidConfigError, "version must not be empty")
+	}
+	if config.Description == "" {
+		return nil, microerror.MaskAnyf(invalidConfigError, "description must not be empty")
+	}
+
+	tpr := &tpr{
+		clientset: config.Clientset,
+
+		name:          config.Name,
+		group:         config.Group,
+		version:       config.Version,
+		description:   config.Description,
+		qualifiedName: config.Name + "." + config.Group,
+
+		endpointList: fmt.Sprintf("/apis/%s/%s/%ss", config.Group, config.Version, config.Name),
+	}
+	return tpr, nil
+}
+
+// CreateAndWait create a TPR and waits till it is initialized in the cluster.
+func (t *tpr) CreateAndWait() error {
+	err := t.create()
+	if err != nil {
+		microerror.MaskAny(fmt.Errorf("creating TPR: %+v", err))
+	}
+	err = t.waitInit()
+	if err != nil {
+		microerror.MaskAny(fmt.Errorf("waiting TPR initialization: %+v", err))
```

Comment:
I would say do either of these. 

```
microerror.MaskAny(err)
```
```
microerror.MaskAnyf(err, "format string %s", "some arg")
```

That way we do not hijack the type masking and can make use of the built in string formatting.